pkg/util/fileconfig.go

```diff
@@ -0,0 +1,257 @@
+    if err != nil {
+        return "", err
+    }
+
+    // #nosec
+    contentRaw, err := ioutil.ReadFile(fPath)
+    if err != nil {
+        return "", err
+    }
+
+    var cephConfig ClusterConfigJSONv1
+
+    err = json.Unmarshal(contentRaw, &cephConfig)
+    if err != nil {
+        return "", err
+    }
+
+    if cephConfig.ClusterConf.ClusterFsID != fsid {
+        return "", fmt.Errorf("mismatching Ceph cluster fsid (%s) in file, passed in (%s)", cephConfig.ClusterConf.ClusterFsID, fsid)
+    }
+
+    if len(cephConfig.ClusterConf.Monitors) == 0 {
+        return "", fmt.Errorf("monitor list empty in configuration file")
+    }
+
+    return strings.Join(cephConfig.ClusterConf.Monitors, ","), nil
+}
+
+// GetProvisionerSubjectID returns the provisioner subject ID from the on-disk
+// configuration file, based on the passed in fsid
+func (fc *FileConfig) GetProvisionerSubjectID(fsid string) (string, error) {
+    fPath, err := provSubject.constructPath(fc.BasePath, fsid)
+    if err != nil {
+        return "", err
+    }
+
+    // #nosec
+    contentRaw, err := ioutil.ReadFile(fPath)
+    if err != nil {
+        return "", err
+    }
+
+    if string(contentRaw) == "" {
+        return "", fmt.Errorf("missing/empty provisioner subject ID from file (%s)", fPath)
+    }
+
+    return string(contentRaw), nil
+}
+
+// GetPublishSubjectID returns the publish subject ID from the on-disk
+// configuration file, based on the passed in fsid
+func (fc *FileConfig) GetPublishSubjectID(fsid string) (string, error) {
+    fPath, err := pubSubject.constructPath(fc.BasePath, fsid)
+    if err != nil {
+        return "", err
+    }
+
+    // #nosec
+    contentRaw, err := ioutil.ReadFile(fPath)
+    if err != nil {
+        return "", err
+    }
+
+    if string(contentRaw) == "" {
+        return "", fmt.Errorf("missing/empty publish subject ID from file (%s)", fPath)
+    }
+
+    return string(contentRaw), nil
+}
+
+// GetCredentialForSubject returns the credentials for the requested subject
+// from the cluster config for the passed in fsid
+func (fc *FileConfig) GetCredentialForSubject(fsid, subject string) (string, error) {
+    var fPath string
+    var err error
+
+    tmpSubject, err := fc.GetPublishSubjectID(fsid)
+    if err != nil {
+        return "", err
+    }
+
+    if tmpSubject != subject {
+        tmpSubject, err = fc.GetProvisionerSubjectID(fsid)
+        if err != nil {
+            return "", err
+        }
+
+        if tmpSubject != subject {
+            return "", fmt.Errorf("requested subject did not match stored publish/provisioner subjectID")
+        }
+
+        fPath, err = provCreds.constructPath(fc.BasePath, fsid)
+        if err != nil {
+            return "", err
+        }
+    } else {
+        fPath, err = pubCreds.constructPath(fc.BasePath, fsid)
+        if err != nil {
+            return "", err
+        }
+    }
+
+    // #nosec
+    contentRaw, err := ioutil.ReadFile(fPath)
```

Comment:
we can make this as function and use with `GetPublishSubjectID` and  here to avoid code duplication